app/models/protocol_subscription.rb

```diff
@@ -78,19 +78,27 @@ def protocol_completion
     responses.map do |response|
       current_streak = -1
 
-      if response.measurement.periodical?
-        on_streak = determine_streak(on_streak, response.completed?, response.still_possible?)
-        current_streak = on_streak
-      end
+      on_streak, current_streak = retrieve_periodical_streak(on_streak, current_streak, response)
 
       create_protocol_completion_entry(response.completed?,
                                        response.measurement.periodical?,
                                        response.measurement.reward_points,
                                        response.still_possible?,
-                                       current_streak)
+                                       current_streak,
+                                       response.future_or_current?)
     end
   end
 
+  def retrieve_periodical_streak(on_streak, current_streak, response)
+    if response.measurement.periodical?
+      on_streak = determine_streak(on_streak, response.completed?, response.still_possible?)
+
+      # Return the on_streak twice, so we update the current_streak
+      return [on_streak, on_streak]
+    end
+    [on_streak, current_streak]
```

Comment:
is het niet wat raar om current_streak als functieargument aan retrieve_periodical_streak mee te geven alssie altijd -1 is?